File: BC-W4/Army/Spells/Fireball/Fireball.cpp
#include "Fireball.h"

Fireball::Fireball(int actionPoints, int manaCost, const std::string& spellName) : Spell(actionPoints, manaCost, spellName) {};
Fireball::~Fireball() {};

void Fireball::action(Unit* enemy) {
    enemy->takeMagicDamage(this->getActionPoints());
};

void Fireball::heal(Unit* unit) {
    throw InvalidHealException();
};

File: BC-W4/ArmyJava/src/spells/Fireball.java
package spells;

import exceptions.InvalidCastException;
import units.Unit;

public class Fireball extends Spell {
    public Fireball() {
        super("Fireball", 40, 60, true);
    }

    public Fireball(int actionPoints) {
        super("Fireball", 40, actionPoints, true);
    }

    public void action(Unit enemy) throws InvalidCastException {
        enemy.takeMagicDamage(this.actionPoints);
    }

}

